### DISCUSSION TITLE
Question about "rendering" / "compiling" email.

### DISCUSSION THREAD

**Discussion by mhelbich** [Dave136/vue-email#167] [Q&A]
Hey all, I'm using this package for the first time - and it seems to work great so far!
I did notice something in my console logs where I couldn't quite figure out whether it was actually a problem or not.
So, my setup is a NestJS API, and we're in a controller.
At the top of the file I import and configure the vue-email package like this:
```ts
import { config } from "@vue-email/compiler";
const vueEmailConfig = config("./src/email", {
    verbose: true, // added this recently, didn't seem to change my log outputs though
});
console.log(vueEmailConfig);
```

In one of my controller endpoints (the usual Nest JS functions), I am using Nest's Email Module to send the rendered vue-email template:
```ts
        // Send registration email
        const template = await vueEmailConfig.render("registration.vue", {
            props: {
                email: body.email,
                token: activationToken,
            },
        });
        const res = await this.mailerService.sendMail({
            to: body.email,
            subject: "Confirm your Registration!",
            text: template.text,
            html: template.html,
        });
        console.log(res);
```

Now the thing that actually confuses me is what I see in my console:
```
🚧 Compiling Registration file
💌 Generating output
🚧 Compiling Registration file
🚧 Compiling Welcome file
🎉 Rendering template Registration
```
I *do* have a Welcome file in that directory as well (the one from the config call), but why is it being "compiled" here? At the moment it's only there to be used later on. At no point in my code am I using this file yet.
Is this normal or am I doing something wrong? :/

Thanks for the replies and I'm sorry if this is a stupid question or caused by something I'm doing wrong. I appreciate anyone reading this! :)

**Comment by Flowko** [comment]
hi, thanks for the feedback, thats actually logs compiling other componetns, cuz currently we dont have a way to know what components are being used in each template, so we render all components, i know, its not the best approach, but till we find a good way to do that, also with logs, did u try to set verbos to false ? it should stopp logging in the console, let me know if that doesnt resolve it, ill look into it

**Reply by mhelbich** [reply]
Hey there, thanks for the quick reply. Good to know about the compiling stuff - I'll try to remember that if we ever do run into performance issues - but it should be fine for now.
As for the console logs... setting `verbose` to falls explicity does actually disable them.
I'll let you know if I ever find a way to prevent the first issue!

**Comment by Flowko** [comment]
we updated the whole library logic, this shouldn;t be a problem anymore, thanks
